### PR TITLE
Add Pull Refresh

### PR DESCRIPTION
**Pull request by noahmalmed**
Addressing #184 

Describe your changes:
- Added pull refresh to the home screen of the app
- Fixed a bug with the google analytics Android dependency
- Some slight refactoring to the projects downloading code

# Review Checklist

- [x] Does it work in Android and iOS?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Are tests passing?

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -172,6 +171,12 @@ dependencies {
     compile 'com.pusher:pusher-websocket-android:0.5.0'
 }
 
+subprojects {
+    if(name.equals('react-native-google-analytics-bridge')) {
```
Comment (wgranger) [inline android/app/build.gradle]: Yes yes yes. I think this should take care of the bug I always experience in Android Studio

Comment (noahmalmed) [reply android/app/build.gradle]: Haha yeah... Not sure why I wasn't getting it until now

**Comment by wgranger** [comment]
Can you double check everything is working in Android Studio? I'm having issues with unresolved dependencies. Failed to resolve: com.facebook.react:react-native:0.52.2

**Comment by wgranger** [comment]
Branch looks good overall but there may have been a dependency issue arising from the use of the caret vs tilde in the package.json causing differences in our builds.

**Pull request merged**